### PR TITLE
e2e framework TestJig used for Service testing wasn't verifying UDP works

### PR DESCRIPTION
**Pull request by aojea**
/kind bug
/kind cleanup

The TestJig helper for Service tests was using `nc -vz -u` to test the UDP service connectivity, however, that command only checks that the packet is sent and doesn't guarantee that the port is open because it never checks the reply.

```
 nc -v -z -u 8.8.8.8 12323 && echo OK
Ncat: Version 7.80 ( https://nmap.org/ncat )
Ncat: Connected to 8.8.8.8:12323.
Ncat: UDP packet sent successfully
Ncat: 1 bytes sent, 0 bytes received in 2.01 seconds.
OK
```

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/framework/service/jig.go [hunk 1]
```diff
@@ -890,19 +882,24 @@ func testEndpointReachability(endpoint string, port int32, protocol v1.Protocol,
...
 	err := wait.PollImmediate(1*time.Second, ServiceReachabilityShortPollTimeout, func() (bool, error) {
-		if _, err := framework.RunHostCmd(execPod.Namespace, execPod.Name, cmd); err != nil {
+		stdout, err := framework.RunHostCmd(execPod.Namespace, execPod.Name, cmd)
+		if err != nil {
 			framework.Logf("Service reachability failing with error: %v\nRetrying...", err)
 			return false, nil
 		}
-		return true, nil
+		trimmed := strings.TrimSpace(stdout)
+		if trimmed != "" {
```
Comment (aojea) [inline test/e2e/framework/service/jig.go:899]: we check that at least we've received something, these checks are duplicated all over the e2e framework, and the `Service Granular Checks` do a more extensive check, verifying that all endpoints belonging to the service reply.
We keep this just with the minimum, checking that at least we receive one reply.

File: test/e2e/framework/service/jig.go [hunk 2]
```diff
@@ -1021,24 +1018,8 @@ func (j *TestJig) CheckServiceReachability(svc *v1.Service, pod *v1.Pod) error {
 	}
 }
 
-// CreateServicePods creates a replication controller with the label same as service. Service listens to HTTP.
+// CreateServicePods creates a replication controller with the label same as service. Service listens to TCP and UDP.
 func (j *TestJig) CreateServicePods(replica int) error {
-	config := testutils.RCConfig{
```
Comment (aojea) [inline test/e2e/framework/service/jig.go:1026]: I didn't see that this was used anywhere, so we can use TCP and UDP directly and simplify the polling logic using netcat only

**Comment by aojea** [comment]
hehe

```
Kubernetes e2e suite: [sig-network] Networking Granular Checks: Services should function for client IP based session affinity: udp [LinuxOnly] expand_more
```
now we are testing it, I was surprised why udp wasn't failing more, and it is that we weren't testing them at all

/retest

**Comment by aojea** [comment]
ping @BenTheElder @spiffxp FYI

File: test/e2e/framework/service/jig.go (hunk 2 above)
Comment (oomichi) [reply test/e2e/framework/service/jig.go:1026]: There are not any existing callers of `CreateServicePods()` which depend on HTTP, +1 for making the function for TCP and UDP.

File: test/e2e/framework/service/jig.go [hunk 3]
```diff
@@ -890,19 +882,24 @@ func testEndpointReachability(endpoint string, port int32, protocol v1.Protocol,
 	cmd := ""
 	switch protocol {
 	case v1.ProtocolTCP:
-		cmd = fmt.Sprintf("nc -zv -t -w 2 %s %v", endpoint, port)
+		cmd = fmt.Sprintf("echo hostName | nc -v -t -w 2 %s %v", endpoint, port)
```
Comment (oomichi) [inline test/e2e/framework/service/jig.go:885]: Just a question: What is a purpose of `echo hostName` ?

**Review by oomichi — COMMENTED** [review]
/cc @oomichi

File: test/e2e/framework/service/jig.go (hunk 3 above)
Comment (aojea) [reply test/e2e/framework/service/jig.go:885]: the pod returns its hostname https://github.com/tallclair/kubernetes/blob/c05a350b4f16bbf1a709b82a97584efd153e6a8e/test/images/agnhost/README.md#netexec

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Pull request merged**